### DISCUSSION TITLE
Integration with heirline

### DISCUSSION THREAD

**Discussion by ryanmsnyder** [linrongbin16/lsp-progress.nvim#60] [Q&A]
In the README, it shows how to integrate with `lualine` with an autocmd:

```lua
-- listen lsp-progress event and refresh lualine
vim.api.nvim_create_augroup("lualine_augroup", { clear = true })
vim.api.nvim_create_autocmd("User LspProgressStatusUpdated", {
    group = "lualine_augroup",
    callback = require("lualine").refresh,
})
```

How can I integrate this with another statusline plugin like [`heirline`](https://github.com/rebelot/heirline.nvim)? I have this right now and though it seems to work, it refreshes my statusline constantly, even without any lsp activity:

```lua
vim.api.nvim_create_augroup("heirline_augroup", { clear = true })
vim.api.nvim_create_autocmd("User LspProgressStatusUpdated", {
  group = "heirline_augroup",
  callback = function()
    vim.cmd "redrawstatus"
  end,
})
```

**Comment by linrongbin16** [comment]
lsp progress will send event constantly  even there's no status updated.

This is because I found statusline won't get updated when first open a lua script (with lua-ls) until my cursor moves, I am not sure why there's no event send by lsp. So I registered a background task, send event every 500 ms.

You can set a very big int to `regular_internal_update_time` field to disable it, see: https://github.com/linrongbin16/lsp-progress.nvim/blob/main/lua/lsp-progress/defaults.lua#L47

**Reply by linrongbin16** [reply]
Hi @ryanmsnyder , now we have a section for integration with heirline: https://github.com/linrongbin16/lsp-progress.nvim?tab=readme-ov-file#heirlinenvim, hope it helps.
